docs/coinListing/README.md

```diff
@@ -0,0 +1,56 @@
+## Coin listing technical architecture :
+
+- make api calls to each exchange for each coin it supports
+
+- parse the response separately for each exchange and coin
+- inject the response into an intermediate coin exchange listing collection (db)
+- aggregate the results of multiple exchanges for each coin through volume weighted average to find the price of coin
+- inject the aggregated price into a coin listing collection which will later be used for analysis
+- repeat the above data fetching steps at regular intervals through cron jobs
+
+## Implementation details :
+
+- Each exchange end point with the coin are fetched from db instead of hardcoding in the code
+
+- Exchange api structure schema
+
+```js
+const exchangeApiStructureSchema = mongoose.Schema({
+  exchange: {
+    type: mongoose.Types.ObjectId,
+    ref: 'Exchange',
+    required: true,
+  },
+  coin: {
+    type: mongoose.Types.ObjectId,
+    required: true,
+    ref: 'Coin',
+  },
+  endPointPath: {
```

Comment:
i guess this is the endpoint path then?

Reply:
This is only the path parameters for the actual end point. Actual end point will be part of exchange schema